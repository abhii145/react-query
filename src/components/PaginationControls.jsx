import PropTypes from "prop-types"

const PaginationControls = ({
  disablePrevious,
  disableNext,
  onMove,
  limit,
}) => (
  <div className="flex justify-between items-center mt-8">
    <button
      onClick={() => onMove(-limit)}
      className={`px-4 py-2 rounded-lg ${
        disablePrevious
          ? "bg-gray-400 text-white cursor-not-allowed"
          : "bg-blue-500 text-black hover:bg-blue-600"
      }`}
      disabled={disablePrevious}
    >
      Previous
    </button>
    <button
      onClick={() => onMove(limit)}
      className={`px-4 py-2 rounded-lg ${
        disableNext
          ? "bg-gray-400 text-white cursor-not-allowed"
          : "bg-blue-500 text-black hover:bg-blue-600"
      }`}
      disabled={disableNext}
    >
      Next
    </button>
  </div>
)

PaginationControls.propTypes = {
  disablePrevious: PropTypes.bool.isRequired,
  disableNext: PropTypes.bool.isRequired,
  onMove: PropTypes.func.isRequired,
  limit: PropTypes.number.isRequired,
}

export default PaginationControls
