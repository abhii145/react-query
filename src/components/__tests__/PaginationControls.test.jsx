import { render, screen, fireEvent } from "@testing-library/react"
import PaginationControls from "../PaginationControls"
import { it, expect, describe, vi } from "vitest"

describe("PaginationControls", () => {
  const onMove = vi.fn()

  it("disables previous button when disablePrevious is true", () => {
    render(
      <PaginationControls
        disablePrevious={true}
        disableNext={false}
        onMove={onMove}
        limit={10}
      />
    )
    const prevButton = screen.getByText("Previous")
    expect(prevButton).toBeDisabled()
  })

  it("disables next button when disableNext is true", () => {
    render(
      <PaginationControls
        disablePrevious={false}
        disableNext={true}
        onMove={onMove}
        limit={10}
      />
    )
    const nextButton = screen.getByText("Next")
    expect(nextButton).toBeDisabled()
  })

  it("calls onMove with correct value when buttons are clicked", () => {
    render(
      <PaginationControls
        disablePrevious={false}
        disableNext={false}
        onMove={onMove}
        limit={10}
      />
    )
    const prevButton = screen.getByText("Previous")
    const nextButton = screen.getByText("Next")
    fireEvent.click(prevButton)
    expect(onMove).toHaveBeenCalledWith(-10)
    fireEvent.click(nextButton)
    expect(onMove).toHaveBeenCalledWith(10)
  })
})
