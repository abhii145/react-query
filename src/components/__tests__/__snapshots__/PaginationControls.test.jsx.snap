// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PaginationControls > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex justify-between items-center mt-8"
  >
    <button
      class="px-4 py-2 rounded-lg bg-blue-500 text-black hover:bg-blue-600"
    >
      Previous
    </button>
    <button
      class="px-4 py-2 rounded-lg bg-blue-500 text-black hover:bg-blue-600"
    >
      Next
    </button>
  </div>
</DocumentFragment>
`;
